Src/Base/AMReX_GpuAtomic.H

```diff
@@ -232,12 +232,12 @@ namespace detail {
 
     template<class T>
     AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
-    T Max (T* m, T value) noexcept
+    T Max (T* const m, T const value) noexcept
```

Comment:
```suggestion
    T Max (std::add_const<T*>::type m, std::add_const<T>::type value) noexcept
```

Reply:
Let's not try to use everything in the standard.  What do we gain?  If `T` is say `double const`, `double const const` will become simply `double const`.  Furthermore, if `T` is `double const`, the code won't compile anyway because `atomicAdd`, `atomicCAS` cannot take `const*` and `retinterpret_cast` cannot cast `const` away.  So why not keeping things clean?